commercetools.Sdk/commercetools.Sdk.Serialization/JsonConverters/AttributesConverter.cs

```diff
@@ -70,9 +72,9 @@ private Type GetAttributeType(JToken value)
                             return typeof(TimeAttribute);
                         case JTokenType.String:
                         {
-                            if (dateTimeAttributeMapper.CanConvert(value))
+                            if (dateTimeAttributeMapper.CanConvert(value) && !serializationConfig.DeserializeDateTimeAttributesAsString)
```

Comment:
i would swap the conditions two avoid unnecessary regex checks if dates will not be deserialized anyway